tests/console/check_package_version.pm

```diff
@@ -0,0 +1,75 @@
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Check package version only for sles12sp5 migration scenarios
+# Maintainer: Yutao Wang <yuwang@suse.com>
+
+use strict;
+use warnings;
+use version;
+use base 'installbasetest';
+use testapi;
+use utils 'systemctl', 'zypper_call';
+
+my @packages = qw(autofs openscap augeas freeradius-server gpgme libgpg-error0); #apache2-mod_auth_openidc);
+my @packagev = qw(5.1.5 1.3.0 1.0.0 3.0.18 1.9.0 1.17);# 0.0.0);
+
+my @packages_s390x = qw(libnuma-devel);
+my @packagev_s390x = qw(2.0.12);
+
+sub search_packages {
+    my ($package) = @_;
+    zypper_call("se -s $package");
+}
+
+sub cmp_version {
+    my ($old, $new) = @_;
+    my @newv = split(/-/, $new);
+    my $v1 = version->parse($old);
+    my $v2 = version->parse($newv[0]);
+    return $v1 <= $v2;
+}
+
+sub cmp_result {
+    my ($pcks, $pckv) = @_;
+    search_packages($pcks);
+    my $output = script_output("zypper se -s $pcks | grep -w $pcks | head -1 | awk -F '|' '{print \$4}'", 80);
+    if ($output ne '' && !cmp_version($pckv, $output)) {
+         record_soft_failure("The $pcks version is $output, but request is $pckv");
+    }
+}
+
+sub run {
+
+    select_console 'root-console';
+    
+    for (my $i = 0; $i < scalar @packages; $i++) {
+        cmp_result($packages[$i], $packagev[$i]);
+    }
+
+    if (get_var('ARCH') =~ /s390x/) {
+        for (my $i = 0; $i < scalar @packages_s390x; $i++) {
+            cmp_result($packages_s390x[$i], $packagev_s390x[$i]);
+        }
+    }
+
+    script_run('python3 --version', 30);
```

Comment:
how python3 version related to whole test ?

Reply:
According https://jira.suse.com/browse/SLE-7100
'A growing number of our customers are using Machine Learning to perform data analysis.  If we want to be a viable platform for these customers, we need to provide the latest python so that these customers can develop using the latest ML tools and run the latest third-party ML apps. '  So I want to check if python3 existed at our system.

Reply:
yeah sure , but the problem is that you not checking anything but just output version.

Reply:
Changed to assert_script_run